### DISCUSSION TITLE
Any way to manually test c++20 features in msvc using this repo's copy of the stl?

### DISCUSSION THREAD

**Discussion by AraHaan** [microsoft/STL#1296] [General] [answered]
Is it currently possible that we can try all of the features in c++ 20 manually within current preview of msvc by cloning this to a c++ project?

I would really love to test out things like the now standard std::format to replace ``boost::format/wformat`` in some of my old code.

By the way anything in the standard to replace the ``BOOST_SCOPE_EXIT`` and ``BOOST_SCOPE_EXIT_END`` macros which I would love to do because I would love to eliminate boost entirely on my mfc program (that oddly enough gets an wincodec.h compile error when it includes it in the mfc headers themselves)?

**Comment by SuperWig** [comment]
Follow the build instructions and then [How To Consume](https://github.com/microsoft/STL#how-to-consume). 
If you follow those instructions and assuming your using a Visual Studio that would mean you go into your project's properties and then VC++ Directories and modify it so it looks a bit like this
![image](https://user-images.githubusercontent.com/2692096/93321066-b4cdc600-f809-11ea-825b-bbd4905d2587.png)

**Comment by CaseyCarter** [comment, marked as answer]
> Is it currently possible that we can try all of the features in c++ 20 manually within current preview of msvc by cloning this to a c++ project?

As you can see from [our Changelog](https://github.com/microsoft/STL/wiki/Changelog), there are currently very few STL differences between 16.8p3 and what's checked-in here. Only the things in the [Expected in VS 2019 16.9 Preview 1](https://github.com/microsoft/STL/wiki/Changelog#expected-in-vs-2019-169-preview-1) are new so far. We made a big push to port C++20 changes into the 16.8p3 release branch a few weeks after the "official" feature-complete deadline - to have more to show off for CppCon - so we're a bit behind where we'd normally be making progress on the next release.

> I would really love to test out things like the now standard std::format to replace `boost::format/wformat` in some of my old code.

@barcharcraz is currently working on `format`, but I don't think we've gotten much checked in yet.

> By the way anything in the standard to replace the `BOOST_SCOPE_EXIT` and `BOOST_SCOPE_EXIT_END` macros which I would love to do because I would love to eliminate boost entirely on my mfc program (that oddly enough gets an wincodec.h compile error when it includes it in the mfc headers themselves)?

WG21-P0052 was progressing to make C++20, but was derailed at the last minute by design concerns that I don't recall right now. I'm not sure about its current status.
